modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/MapBoxStyle.java

```diff
@@ -87,15 +88,29 @@ public static StyledLayerDescriptor parse(InputStream stream) throws IOException
             return problems;
         }
         MBStyleTransformer transform = new MBStyleTransformer();
-        
-        for (MBLayer layer : style.layers()) {
-            try {
-                FeatureTypeStyle featureTypeStyle = transform.transform(layer, style);
+        if( style.layers().isEmpty()){
+            problems.add( new MBFormatException("No layers defined"));
+        }
+        else {
+            boolean hasVisibleLayer = false;
+            for (MBLayer layer : style.layers()) {
+                if(!layer.visibility() ){
+                    continue;
+                }
+                try {
+                    FeatureTypeStyle featureTypeStyle = transform.transform(layer, style);
+                    if( featureTypeStyle != null ){
```

Comment:
thinking about this now we could create a featureTypeStyle with Filter.EXCLUDES to represent visibility NONE